app/master/build_request_handler.py

```diff
@@ -107,110 +100,13 @@ def _prepare_build_async(self, build, project_lock):
             analytics.record_event(analytics.BUILD_PREPARE_START, build_id=build.build_id(),
                                    log_msg='Build preparation loop is handling request for build {build_id}.')
             try:
-                self._prepare_build(build)
+                build.prepare(self._subjob_calculator)
```

Comment:
I recognize the benefit (especially in testing) of injecting dependencies like this, instead of instantiating them directly inside of the method itself.

I'm not asking you to change this or anything, but should we make this pattern the norm?